src/textual/widgets/_tabbed_content.py

```diff
@@ -70,6 +73,26 @@ def __init__(
         )
 
 
+class AwaitTabbedContent:
+    """An awaitable return by [`TabbedContent`][textual.widgets.TabbedContent] methods that modify the tabs."""
+
+    def __init__(self, *awaitables: AwaitMount | AwaitRemove) -> None:
+        """Initialise the awaitable.
+
+        Args:
+            *awaitables: The collection of awaitables to await.
+        """
+        super().__init__()
+        self._awaitables = awaitables
+
+    def __await__(self) -> Generator[None, None, None]:
+        async def await_tabbed_content() -> None:
+            for awaitable in self._awaitables:
```

Comment:
Suspect we should use `asyncio.gather` here. Effect is the same, but it will be a little more efficient.

Reply:
What sort of efficiency gain will we see here, and will it guarantee the order that they're awaited?

Reply:
I suspect that `asyncio.gather` _doesn't_ guarantee waiting order.

Reply:
If these are `AwaitMount` and `AwaitRemove` objects, then they are running concurrently anyway. The order you await them won't change the order they are completed.

`gather` will await multiple things with less context switches. Doubt it is a tangible difference for 2, but it is the prefered way to await concurrent tasks.

Reply:
> If these are AwaitMount and AwaitRemove objects, then they are running concurrently anyway. The order you await them won't change the order they are completed.

Ahh, interesting; this is a possible wrinkle to worry about here then. Might need to rethink this perhaps.